core/new-gui/src/app/workspace/types/workflow-common.interface.ts

```diff
@@ -17,12 +17,27 @@ export interface OperatorPort
     portID: string;
   }> {}
 
+export type PartitionInfo =
+  | Readonly<{ type: "hash"; hashColumnIndices: number[] }>
+  | Readonly<{ type: "range"; rangeColumnIndices: number[]; rangeMin: number; rangeMax: number }>
+  | Readonly<{ type: "single" }>
+  | Readonly<{ type: "none" }>;
+
+export interface PortSchema
+  extends Readonly<{
+    jsonSchema: Readonly<JSONSchema7>;
+  }> {}
+
+export interface PortProperty extends Readonly<{ partitionInfo: PartitionInfo; dependencies: number[] }> {}
+
 export interface PortDescription
   extends Readonly<{
     portID: string;
     displayName?: string;
     allowMultiInputs?: boolean;
     isDynamicPort?: boolean;
+    partitionRequirement?: PartitionInfo;
+    dependencies?: number[];
```

Comment:
isn't these two also `PortProperty`? try to simplify those objects and reuse as much as possible?